### ISSUE TITLE
Error on reload the container

### ISSUE THREAD

**Issue by ThomasSousa96** [eclipse-ee4j/jersey#4372]
Hello!

I'm using HotswapAgent to enable the hot-reload. This library has a plugin to supports Jersey. This plugin basically invokes the [`org.glassfish.jersey.servlet.ServletContainer.reload()`](https://github.com/HotswapProjects/HotswapAgent/blob/8f1d97cfc1c0ba78bb2dfdc4145082e2f815bb8a/plugin/hotswap-agent-jersey2-plugin/src/main/java/org/hotswap/agent/plugin/jersey2/Jersey2Plugin.java#L184-L198) method.

When it invokes this method the application throws an exception.

See the issue for more details:

https://github.com/HotswapProjects/HotswapAgent/issues/333

**Comment by jansupol** [comment]
Yes, this is definitely a bug which we experience. It needs to be fixed.
